channel.py

```diff
@@ -287,10 +300,21 @@ def recv_mb(self):
             The list of numpy arrays for the minibatch
 
         """
-        headers = self.asocket.recv_json()
+        socks = dict(self.apoller.poll(self._socket_timout))
+        if socks:
+            if socks.get(self.asocket) == zmq.POLLIN:
+                headers = self.asocket.recv_json()
+        else:
+            raise Exception("Batch socket: recv_json timeout")
+
         arrays = []
         for header in headers:
-            data = self.asocket.recv(copy=False)
+
+            if socks and socks.get(self.asocket) == zmq.POLLIN:
+                data = self.asocket.recv(copy=False)
+            else:
+                raise Exception("Batch socket: recv timeout")
+
```

Comment:
Good point but, actually shouldn't I redo `socks = dict(self.apoller.poll(self._socket_timout))` instead?
In case the master died between the 2 calls.